examples/transforms/transform.rs

```diff
@@ -0,0 +1,113 @@
+use bevy::prelude::*;
+
+use std::f32::consts::PI;
+
+struct CubeState {
+    start_pos: Vec3,
+    move_speed: f32,
+    turn_speed: f32,
+}
+
+struct Center {
+    min_size: f32,
+}
+
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(move_cube.system())
+        .add_system(rotate_cube.system())
+        .add_system(scale_sphere_proportional_to_cube_travel_distance.system())
+        .run();
+}
+
+fn move_cube(mut cubes: Query<(&mut Transform, &mut CubeState)>, timer: Res<Time>) {
+    for (mut transform, cube) in cubes.iter_mut() {
+        let forward = transform.forward();
+        transform.translation += forward * cube.move_speed * timer.delta_seconds();
+    }
+}
+
+fn rotate_cube(mut cubes: Query<(&mut Transform, &mut CubeState)>, timer: Res<Time>) {
+    for (mut transform, cube) in cubes.iter_mut() {
+        let full_turn = transform
+            .looking_at(Vec3::ZERO, transform.local_y())
+            .rotation;
+        let incremental_turn_weight = cube.turn_speed * timer.delta_seconds();
+        transform.rotation = transform.rotation.lerp(full_turn, incremental_turn_weight);
+    }
+}
+
+fn scale_sphere_proportional_to_cube_travel_distance(
+    mut transformabels: QuerySet<(
+        Query<(&Transform, &CubeState)>,
+        Query<(&mut Transform, &Center)>,
+    )>,
+) {
+    let distance: f32 = transformabels
+        .q0()
+        .iter()
+        .map(|(transform, cube)| (cube.start_pos - transform.translation).length())
+        .fold(0.0, |x, y| x + y);
+    for (mut transform, center) in transformabels.q1_mut().iter_mut() {
+        let new_size = Vec3::ONE - 0.05 * Vec3::ONE * distance;
+        let min_size = center.min_size * Vec3::ONE;
+        transform.scale = if new_size.length() < min_size.length() {
```

Comment:
If you don't need the result of `length` and are only performing a comparison then `length_squared` is much cheaper as it avoids a `sqrt`.

Reply:
I think I got it even simpler by just using the scalar values, since I was uniformly scaling, and added that bit just to prevent overly small (or negative sized) spheres. We'll see once I've pushed the update, I guess...